ntbs-service/Services/TreatmentOutcomeService.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System.Collections.Generic;
+using System.Data;
+using System.Data.SqlClient;
+using System.Linq;
+using System.Threading.Tasks;
+using Castle.Core.Internal;
+using Dapper;
+using Microsoft.EntityFrameworkCore.Internal;
+using Microsoft.Extensions.Configuration;
+using ntbs_service.DataAccess;
+using ntbs_service.Models;
+using ntbs_service.Models.Entities;
+using ntbs_service.Models.Enums;
+using ntbs_service.Models.ReferenceEntities;
+
+namespace ntbs_service.Services
+{
+    public interface ITreatmentOutcomeService
+    {
+        bool IsTreatmentOutcomeNeededAtXYears(Notification notification, int yearsAfterTreatmentStartDate);
+        TreatmentOutcome GetTreatmentOutcomeAtXYears(Notification notification, int yearsAfterTreatmentStartDate);
+    }
+
+    public class TreatmentOutcomeService : ITreatmentOutcomeService
+    {
+        public TreatmentOutcomeService()
+        {
+        }
+
+        public TreatmentOutcome GetTreatmentOutcomeAtXYears(Notification notification, int yearsAfterTreatmentStartDate)
+        {
+            // If a treatment outcome is needed at X years then one must not already exist
+            if (IsTreatmentOutcomeNeededAtXYears(notification, yearsAfterTreatmentStartDate))
+            {
+                return null;
+            }
+            
+            // If a treatment outcome is not needed that is because one either exists as the last event of the 1 year period
+            // or one is not needed and so is null
+            return GetOrderedTreatmentEventsInWindowXtoXMinus1Years(notification, yearsAfterTreatmentStartDate)?.First(x => x.TreatmentOutcome != null).TreatmentOutcome;
+        }
+
+        public bool IsTreatmentOutcomeNeededAtXYears(Notification notification, int yearsAfterTreatmentStartDate)
+        {
+            for (var i = yearsAfterTreatmentStartDate; i >= 1; i--)
+            {
+                var lastTreatmentEventsBetweenIAndIMinusOneYears = GetOrderedTreatmentEventsInWindowXtoXMinus1Years(notification, i)?.FirstOrDefault();
```

Comment:
Reference to First here, but Last in variable name seems a little disjointed - I would probably just order by asc, and .LastOrDefault() here